client_secrets.ps1

```diff
@@ -178,44 +178,52 @@ try {
               Remove-AzADAppCredential -ObjectId $objectId -KeyId $keyId -DefaultProfile $targetContext
             }
             elseif ($expiringRangeDateObj.Days -lt $expiringRangeDays) {
-              Write-Output "$keyName will expire within $expiringRangeDays."           
+              Write-Output "$keyName will expire within $expiringRangeDays."
             }
             else {
               Write-Output "$kvSecretName secret is not expiring"
               $validSecrets = $true
             }
-           
+
           }
 
           if (!$validSecrets) {
             Write-Output "There are no valid secrets"
-    
+
             Write-Output "Creating Secret $kvSecretName"
             $passCreds = [PSCustomObject]@{
               DisplayName = $displayName
               EndDateTime = $secretEndDate
               StartDateTime = $secretStartDate
             }
             $response = New-AzADAppCredential -ObjectId $objectId -PasswordCredentials $passCreds -DefaultProfile $targetContext
-    
-            ## Add/Update Secret 
+
+            ## Add/Update Secret
             $secretvalue = ConvertTo-SecureString $response.secretText -AsPlainText -Force
             Set-AzKeyVaultSecret -VaultName $key_vault_name -Name "$kvSecretName-pwd" -SecretValue $secretvalue -DefaultProfile $sourceContext
           }
 
           Write-Output "Recycling $appName Secrets ENDED"
         }
 
-        Write-Output "Saving ID to $key_vault_name"
-        $secretvalue = ConvertTo-SecureString $appId -AsPlainText -Force
-        Set-AzKeyVaultSecret -VaultName $key_vault_name -Name "$kvSecretName-id" -SecretValue $secretvalue -DefaultProfile $sourceContext
+        try {
```

Comment:
I wonder if this bit should check whether the value matches rather than the key exists.

It's an unlikely scenario, but technically could happen if someone re-create the app reg so I think it'd be worth sorting now